lib/algrep.gi

```diff
@@ -274,11 +274,11 @@ true,
       local F,V;
 
       if str<>"basis" then
-         Error( "Usage: RightAlgebraModule( <A>, <op>, <gens>, <str>) where the last argument is the string \"basis\"" );
+         Error( "Usage: BiAlgebraModuleByGenerators( <A>, <B>, <opl>, <opr>, <gens>, <str>) where the last argument is the string \"basis\"" );
       fi;
 
       F:= LeftActingDomain( A );
-      V:= VectorSpace( F, str );
+      V:= VectorSpace( F, gens );
```

Comment:
This looks like a behavior change (and a good one). It would be good to add a test that verifies it. E.g. by adapting one of the existing tests in the manual examples:
```
A:= FullMatrixAlgebra( Rationals, 3 );;
V:= BiAlgebraModuleByGenerators( A, A, \*, \*, [ [1,0,0] ], "basis" );
```

Reply:
Done. I still don't see the logic of behind having a function that only accepts a fixed string "basis" as its last argument, especially when there is a 5 argument version of `BiAlgebraModuleByGenerators` (immediately before the method modified here), which is identical to the 6 argument version except for the check that the last argument is `"basis"`.

Reply:
I think what happened here is that someone took a "regular" vararg function that has an optional final argument `"basis"` (e.g. `VectorSpace`, `Subspace`) and decided to turn it into methods for an operation -- and methods can be variadic.

Reply:
Sure, the optional last argument has no impact on the method, and so I wonder why it exists, maybe it's used elsewhere where there are more options? 

Anyway, I just de-duplicated the code, otherwise, I think I'm done with this PR, unless there are further changes suggested.